tutorials/ann_hnswlib.ipynb

```diff
@@ -0,0 +1,394 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "b9a4225b-1a05-4b58-9e1d-1511650ef225",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Requirement already satisfied: hnswlib in /opt/conda/lib/python3.10/site-packages (0.7.0)\n",
+      "Requirement already satisfied: numpy in /opt/conda/lib/python3.10/site-packages (from hnswlib) (1.26.0)\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip install hnswlib"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "74a9e78f-3e8a-4ee2-89fe-b3a3f4784b53",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import hnswlib\n",
+    "import cornac\n",
+    "from cornac.data import Reader\n",
+    "from cornac.datasets import netflix\n",
+    "from cornac.eval_methods import RatioSplit\n",
+    "from cornac.models import MF, HNSWLibANN"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cf6bb9a5-ffb5-4221-8122-9aa286af1d9c",
+   "metadata": {},
+   "source": [
+    "## Recommender model training\n",
+    "\n",
+    "We do a small experiment to demonstrate how Approximate Nearest Neighbor (ANN) search could be performed in Cornac. First, we need to train a model that supports ANN search. Here we choose MF for simple illustration purpose. Other models that support ANN search should work in a similar fashion."
```

Comment:
"We do a small experiment to demonstrate how Approximate Nearest Neighbor (ANN) search could be performed in Cornac" --> The following experiment shows how to perform ANN-search within Cornac

Reply:
fixed